matrix_sdk/src/room/common.rs

```diff
@@ -0,0 +1,104 @@
+use matrix_sdk_common::api::r0::{
+    membership::{forget_room, get_member_events, join_room_by_id, leave_room},
+    message::get_message_events,
+};
+use std::ops::Deref;
+
+use crate::{Client, Result, Room, RoomMember};
+
+/// A struct containing methodes that are common for Joined, Invited and Left Rooms
+#[derive(Debug, Clone)]
+pub struct Common {
+    inner: Room,
+    pub(crate) client: Client,
+}
+
+impl Deref for Common {
+    type Target = Room;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl Common {
+    /// Create a new `room::Common`
+    ///
+    /// # Arguments
+    /// * `client` - The client used to make requests.
+    ///
+    /// * `room` - The underlaying room.
+    pub fn new(client: Client, room: Room) -> Self {
```

Comment:
This shouldn't be pub, no?

Reply:
I think the user may need to construct the `room::{Common, Joined, Invited, Left}` especially since the `EventHandler` returns only the `matrix_sdk_base::Room`.

Reply:
Yeah, that's why we'll want to move the `EventHandler` into the `matrix-sdk` and it should emit those higher level rooms as well.

Reply:
What are you suggestion to do for now, so we don't break the sdk till we move the `EventHandler`?

Reply:
I guess we can leave this `pub` for now, just add an TODO. That should end up being the least amount of work until the `EventHandler` moves.

Reply:
Sounds good :)